src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/mailer.html.twig

```diff
@@ -101,135 +63,184 @@
+                                                            src="data:text/html;charset=utf-8;base64,{{ collector.base64Encode(htmlBody) }}"
+                                                            style="height: 80vh;width: 100%;"
+                                                        >
+                                                        </iframe>
+                                                    </pre>
                                                 </div>
                                             </div>
+                                        {% endif %}
+
+                                        {% if message.textBody %}
+                                            {% set textBody = message.textBody() %}
                                             <div class="tab">
-                                                <h3 class="tab-title">Raw</h3>
+                                                <h3 class="tab-title">Text content</h3>
                                                 <div class="tab-content">
-                                                    <pre class="prewrap" style="max-height: 600px">{{ message.toString() }}</pre>
+                                                    <pre class="prewrap" style="max-height: 600px">
+                                                        {%- if message.textCharset() %}
+                                                            {{- textBody|convert_encoding('UTF-8', message.textCharset()) }}
+                                                        {%- else %}
+                                                            {{- textBody }}
+                                                        {%- endif -%}
+                                                    </pre>
                                                 </div>
                                             </div>
-                                        </div>
+                                        {% endif %}
                                     </div>
-                                {% endif %}
-                            </div>
+                                </div>
+                            {% endif %}
                         </div>
                     </div>
-                {% endfor %}
-            </div>
+
+                    <div class="tab">
+                        <h3 class="tab-title">MIME parts</h3>
+                        <div class="tab-content">
+                            <pre class="prewrap" style="max-height: 600px; margin-left: 5px">{{ message.body().asDebugString() }}</pre>
+                        </div>
+                    </div>
+
+                    <div class="tab">
+                        <h3 class="tab-title">Raw Message</h3>
+                        <div class="tab-content">
+                            <a class="mailer-message-download-raw" href="data:application/octet-stream;base64,{{ collector.base64Encode(message.toString()) }}" download="email.eml">
```

Comment:
small thing, if doable
perhaps add some unitary string data into the `email.eml` to have different names in case of multi email?

Reply:
I'd say we shouldn't use the email subject (because it can be too long; it can contain problematic characters for file names, which we cannot slugify without introducing other issues, etc.)

Given that all browsers auto-append a numeric suffix when downloading the same filename ... when having multiple emails you'll end up with `email.eml`, `email (1).eml`, `email (2).eml`, etc.

I think it's OK for this feature, which will be mostly used as: click to download email; double click downloaded file to open in an email client; check that it looks OK; delete the file.